internal/providercache/dir.go

```diff
@@ -42,6 +46,13 @@ type Dir struct {
 	// directory made by other codepaths because the contract for NewDir
 	// explicitly defines using the same directory for multiple purposes
 	// as undefined behavior.
+	// However, this code is now used for the global provider cache. With
+	// the added support for locking, the data may no longer be valid with
+	// changes from other processes. In practice this means that some packages
+	// may have been installed since the latest re-scan. The code that
+	// handles the installation should be smart enough to detect that and
+	// work around it.
+
```

Comment:
nit: there seems to be an errant blank link at the end of this comment now, which would probably cause refactoring tools to not realize that this comment belongs to the `metaCache` item.